packages/synapse-react-client/src/components/DataGrid/DataGrid.tsx

```diff
@@ -69,17 +68,15 @@ const DataGrid = () => {
 
   // Grid rows and columns
   type DataGridRow = { [key: string]: string | number }
-  const [rowValues, setRowValues] = useState<DataGridRow[]>(
-    modelRowsToGrid(modelSnapshot),
-  )
-  const [colValues, setColValues] = useState<Column[]>(
-    modelColsToGrid(modelSnapshot),
-  )
+  const [rowValues, setRowValues] = useState<DataGridRow[]>([])
+  const [colValues, setColValues] = useState<Column[]>([])
   const [prevRows, setPrevRows] = useState(rowValues)
 
   useEffect(() => {
-    setRowValues(modelRowsToGrid(modelSnapshot))
-    setColValues(modelColsToGrid(modelSnapshot))
+    if (modelSnapshot) {
+      setRowValues(modelRowsToGrid(modelSnapshot))
+      setColValues(modelColsToGrid(modelSnapshot))
+    }
   }, [modelSnapshot])
```

Comment:
Updated `modelSnapshot` so it may be null. So initialize the rows/cols to empty arrays and don't update them using a null snapshot.